logstash-core/src/main/java/org/logstash/util/PasswordValidator.java

```diff
@@ -0,0 +1,128 @@
+package org.logstash.util;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+
+import java.util.List;
+
+/**
+ * A class to validate the given password string and give a reasoning for validation failures.
+ * Validation policies are based on complex password generation recommendation from several institutions
+ * such as NIST (https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-63b.pdf),
+ * OWASP (https://github.com/OWASP/www-community/blob/master/pages/OWASP_Validation_Regex_Repository.md), etc...
+ */
+public class PasswordValidator {
+
+    /**
+    A regex for special character inclusion.
+     */
+    private static final String SYMBOL_REGEX = ".*[~!@#$%^&*()_+|<>?:{}].*";
+
+    /**
+     A regex for upper case character inclusion.
+     */
+    private static final String UPPER_CASE_REGEX = ".*[A-Z].*";
+
+    /**
+     A regex for lower case character inclusion.
+     */
+    private static final String LOWER_CASE_REGEX = ".*[a-z].*";
+
+    /**
+     A regex for digit number inclusion.
+     */
+    private static final String DIGIT_REGEX = ".*\\d.*";
+
+    /**
+     A policy failure reasoning for empty password.
+     */
+    private static final String EMPTY_PASSWORD_REASONING = "Password must not be an empty.";
+
+    /**
+     A policy failure reasoning for password length.
+     */
+    private static final String LENGTH_REASONING = "Password must be length of between 8 and 20.";
+
+    /**
+     A policy failure reasoning if password does not contain special character(s).
+     */
+    private static final String SYMBOL_REASONING = "Password must contain at least one special character.";
+
+    /**
+     A policy failure reasoning if password does not contain upper case character(s).
+     */
+    private static final String UPPER_CASE_REASONING = "Password must contain at least one upper case.";
+
+    /**
+     A policy failure reasoning if password does not contain lower case character(s).
+     */
+    private static final String LOWER_CASE_REASONING = "Password must contain at least one lower case.";
```

Comment:
Example of refactoring according to the [suggestion above](https://github.com/elastic/logstash/pull/14000/files#diff-291759c068472d4dd44be74d9d757c239e6e457b105df0f5d58218cc6a2b507fR558):

```suggestion
    private static final String LOWER_CASE_REASONING = "must contain at least one lower case";
```